wgpu-rs/src/lib.rs

```diff
@@ -501,9 +501,17 @@ impl Device {
         let type_size = std::mem::size_of::<T>() as u32;
         assert_ne!(type_size, 0);
 
-        let desc = BufferDescriptor {
-            size: type_size * count as u32,
-            usage,
+        let desc = if count == 0 {
+            BufferDescriptor {
+                size: 1,
+                usage,
+            }
+        }
+        else {
+            BufferDescriptor {
+                size: type_size * count as u32,
```

Comment:
can we move the `if count == 0` branch deeper into the `size: ` computation? We should probably just do `(size_type * count).max(1) as u32`

Reply:
All good :)